### DISCUSSION TITLE
Colors for numbers, arrows and graphs.

### DISCUSSION THREAD

**Discussion by SideTipeOne** [j-kaltes/Juggluco#194] [Ideas]
Hello everyone.

First of all I wanted to thank you for this very useful App, and this great intuition.
I hope this project can improve more and more, and be able to satisfy the ever-increasing needs of people with diabetes.

I wanted to clarify and suggest something that I didn't find on the App, but that I would find really useful.
After a some time of using the Juggluco App (on SamsungS24, Juggluco version "8.1. 5-phone-li3-si"), and browsing the settings features offered by the App, I noticed that it is not possible to set colors for the numbers, arrows, or portions of graphs, based on their trend.
I believe it is not actually provided for by the App because the only interventions on color customization offered are that of the graphs (average and min per min) and the "floating glucose".
However, I would personally find it really useful, and I don't know how many users can say the same (it's just my idea), to be able to have the sensor glucose numbers, the trend arrows and the graph that are colored based on the value they return.
Since I find the "color code" feedback really a very useful additional feature to pay attention or not to the patient in the follow up of his values ​​and his trend.
Often we do not pay attention to the precise number, due to lack of time, so having a color code that immediately suggests the state of the situation is really useful for me. 

Maybe we could think about implementing the possibility of offering a starting default that can be good for all users as happens with all glucose control apps with sensors: with values ​​that from 70mg/dl to 180mg/dl follow a "green" color for example, and can gradually change color once established thresholds are exceeded that are common to everyone, and in the same way the portion of the graph involved and the arrow near the value. 
So going upwards it could be a "yellow" color if the value exceeds 180 up to 250, "orange" from 250 up to 350, and "dark orange" or other, maybe "purple" beyond that value;
While going downwards: a simple "red" if the glucose exceeds the value 70.
This could be an acceptable default starting function, while still being able to continue to leave the user the possibility of customizing his interface of the colors (most congenial to him), his most optimal range of values ​​(and in which color to see them), in a specific section.

I don't know if it is difficult to implement a similar feature, and identify a section on the App (maybe I'm thinking of an "advanced settings" section) to leave the discretion and needs of the user the possibility of being able to activate or deactivate it (if not wanted and you want to continue using it as it already works now). Maybe through a check, which if clicked, from there a special menu opens in which you can also customize your "color code" as you like based on the values, both for the numbers, and for the trends (so arrows) and the graphs. 

I am not familiar and skilled with computer language, and I do not know if it is possible to implement a similar functionality with simplicity without compromising the philosophy of essentiality that I think the App is inspired by, but I think it could be very useful for the user experience to be able to have feedback with their own color codes. 
And I think this could also improve the graphic interface of the App. 

I personally am tempted to see it as a simple thing to do, but I don't know if its absence is justified by some right (and better reasoned) motivation. Also because I don't think I'm the only one who thought so, I don't know...😇

Thank you for your attention and I look forward to reading your points of view and in case you find this idea useful. 

🙂💪🏻💪🏻
Stay in TIR!

**Comment by franchab** [comment]
Hi everybody!
This is a good suggestion. Before installing Juggluco yesterday I've tried Glucodatahandler whose widget shows value and arrow in different color according to the glucose level. Would be nice if it were the same with Juggluco for the widgets (floating or not) and the glucose value in the state bar (please, at least use a bit larger characters, I'm 71 and it's difficult for me to read the value.
I like very much Juggluco, first for it's direct connexion with my captor. Congratulations!
